compiler/rustc_ast/src/ast.rs

```diff
@@ -1450,11 +1464,13 @@ pub enum ExprKind {
     /// An underscore, used in destructuring assignment to ignore a value.
     Underscore,
 
+    // njn: update this comment
     /// Variable reference, possibly containing `::` and/or type
     /// parameters (e.g., `foo::bar::<baz>`).
     ///
     /// Optionally "qualified" (e.g., `<Vec<T> as SomeTrait>::SomeType`).
-    Path(Option<P<QSelf>>, Path),
+    Path1(Path),
+    Path2(P<QSelf>, P<Path>),
```

Comment:
Maybe `Path` and `QualifiedPath` are more descriptive names than `Path1` and `Path2`?

Reply:
This is draft code, I will clean it up if the perf results indicate it's worth merging.